### DISCUSSION TITLE
BLTouch Z axis doesn't go down

### DISCUSSION THREAD

**Discussion by GeorgeACJ** [Jyers/Marlin#1293] [Q&A] [answered]
Installed the E3V2-BLTouch-3x3-HS-v4.2.2 having an issue where the Z axis when homing only rises on the Z axis and never goes down.

**Comment by Aupuma** [comment]
Same problem here, it was working until recently I disconnected the some motors to fix some stuck filament and now it doesn't work anymore. When homing it moves the probe but then doesn't move the Z axis and gives an error. On Octoprint it says the following:

Recv: Error:!! STOP called because of BLTouch error - restart with M999
Changing monitoring state from "Operational" to "Error"

**Reply by agarza647** [reply]
Similar problem. I was using 1.3.3 then upgraded to 1.3.5 and noticed the Z-axis not moving down anymore. Similar to Aupuma, I had replaced the stock extruder with a dual gear extruder which entailed removing and disconnecting the motor.
I have flashed back to Creality's firmware and the printer homes as normal. Re-flash to Jyers 1.3.3, same problem as before.

**Reply by agarza647** [reply]
Confirmed: motherboard is 4.2.2 and using Jyers for 4.2.2

**Reply by agarza647** [reply]
@tititopher68-dev
So I can just swap the white and black on the 5-pin connector to fix things?

I have never compiled firmware so it would be something new to learn.

**Reply by agarza647** [reply]
I thought you might say that, so I started looking into what I would need for compiling firmware.
I have started looking at YouTube vids and did see that one. I will update once I get some time to install a modified firmware and found a solution.

**Reply by antaenc** [reply]
I had this after building ender extender kit, couldn't see it but one of the wires in the - in my case - CR Touch cable had separated inside the outer coating and there was no continuity.

**Reply by agarza647** [reply]
I have changed the Z_MIN_PROBE_ENDSTOP_INVERTING to true in `Configuration.h`. This caused an error that would not let complete compiling. So I traced the issue to an if statement in `Conditionals_LCD.h` and I commented out the `#error` line. This allowed the compile to complete successfully. I flashed the firmware to my Ender 3v2.

Now another problem has occurred but I am not sure if it is because of the ENDSTOP_INVERTING. When I go to **Home ALL**, the X and Y-axes home properly, the BLTouch pin extends but when the probe touches the bed the Z-axis continues going down and I have to manually cut the power so the nozzle doesn't dig into the bed.

What is causing this? How do I fix it?

**Reply by agarza647** [reply]
@tititopher68-dev
Here is my color order:

- G = Brown
- V = Red
- In = Yellow
- G = Black
- Out = White

**Reply by agarza647** [reply]
@tititopher68-dev 
I had been planning on re-doing the connectors on my BLTouch cabling because it allowed for problems to arise. So I decided to replace the Dupont and JTX connectors and recheck the wiring while I was doing this. I connected the BLTouch to the mainboard temporarily to check for functionality. I powered on the printer and the sensor worked on startup. I flashed back to Creality's firmware and the printer Homed all axes. I flashed Jyers firmware again and the printer Homed all axes properly.

It seems that there must have been a problem with the wiring in the first place. I made sure to hot glue the connection on the mainboard and to zip-tie the extension cable to the sensor.

I have checked all my settings/parameters and everything is working as before. Thanks for the help.

**Comment by antaenc** [comment]
I had this after building ender extender kit, couldn't see it but one of the wires in the - in my case - CR Touch cable had separated inside the outer coating and there was no continuity.

**Comment by mechnut450** [comment]
So i tried this  firmware with great failure lol ( i have a Tresbro knock off I am guessing probe) but have same issues as others on my ender 3v2 with the flash i had to go back to the original flash to get the probe to work now testing a 6 hour print with original firmware..  I was so looking forward to the features that was offered in the jyers firmware.

**Reply by Jyers** [reply]
Make sure you reset eeprom after flashing. (Control menu, reset to defaults then store settings) Not doing so can cause many issues

**Reply by mechnut450** [reply]
actually did this. i even went back to  creality site and got their flash ( with bl touch ) and it still didn't work i had to put the original flash back on from the touch supplier  to get it to work.

**Reply by Jyers** [reply]
Perhaps you have the probe connected differently. It should be connected to the 5 pin BL Touch port on the mainboard

**Comment by Jyers** [comment, marked as answer]
Going to mark this as solved as it's branched a lot. If any of you still have issues perhaps start new discussion threads.
